### ISSUE TITLE
Modal - focus is not trapped inside the last opened modal in 'modal inside modal' case

### ISSUE THREAD

**Issue by mihalichzh** [semrush/intergalactic#712]
**Describe the bug**
Noticed, that focus trap doesn't work properly for 'modal inside modal' case

**To Reproduce**
1. Open modal from another modal ([this](https://developer.semrush.com/intergalactic/components/modal/modal-code/#modal_window_inside_a_modal_window) example from docs can be used)
2. Focus over elements in the last opened modal

**Expected behavior**
Focus is trapped inside the last opened modal

**Additional context**
Focus is not trapped and leaks to the previous modal (see att. screencast).

https://github.com/semrush/intergalactic/assets/39348763/4f87f218-1347-43c1-a208-dab52cd30ed2
